pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+		/// Removes an existing sending account/currency delay
+		pub fn remove_allowance_delay(
+			origin: OriginFor<T>,
+			currency_id: CurrencyIdOf<T>,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			match <AccountCurrencyDelay<T>>::get(&account_id, &currency_id) {
+				Some(_) => {
+					<AccountCurrencyDelay<T>>::remove(&account_id, &currency_id);
+					Self::deposit_event(Event::TransferAllowanceDelayRemoval {
+						sender_account_id: account_id,
+						currency_id,
+					});
+					Ok(())
+				}
+				None => {
+					Self::deposit_event(Event::TransferAllowanceDelayRemoval {
+						sender_account_id: account_id,
+						currency_id,
+					});
+					Err(DispatchError::from(Error::<T>::NoMatchingDelay))
+				}
+			}
+		}
+	}
+
+	impl<T: Config> Pallet<T>
+	where
+		<T as frame_system::Config>::AccountId: Into<Location>,
+	{
+		/// Increments number of allowances present for a sending account/currency set.
+		/// If no allowances set, an entry with 1 added, if entry already present, it is then incremented.
+		pub fn increment_or_create_allowance_count(
+			account_id: &T::AccountId,
+			currency_id: &T::CurrencyId,
+		) -> DispatchResult {
+			// not using try_mutate here as we're not sure if key exits, and we're already doing a some value check on result of exists query check
+			match <AccountCurrencyTransferRestriction<T>>::get(account_id, currency_id) {
+				Some(allowance_count) if allowance_count > 0 => {
+					let new_allowance_count = allowance_count
+						.checked_add(1)
+						.ok_or(Error::<T>::AllowanceCountOverflow)?;
+					<AccountCurrencyTransferRestriction<T>>::insert(
+						account_id,
+						currency_id,
+						new_allowance_count,
+					);
+					Ok(())
+				}
+				Some(_) => Err(DispatchError::from(
+					Error::<T>::AllowanceCountArithmeticError,
+				)),
+				_ => {
+					<AccountCurrencyTransferRestriction<T>>::insert(account_id, currency_id, 1);
+					Ok(())
+				}
+			}
+		}
+
+		/// Decrements the number of allowances tracked for a sending account/currency set.
+		/// If the allowance count is currently 1, then it removes the entry
+		/// If greater than 1, then decremented.
+		/// If no entry present, NoAllowancesSet error returned.
+		pub fn decrement_or_remove_allowance_count(
+			account_id: &T::AccountId,
+			currency_id: &T::CurrencyId,
+		) -> DispatchResult {
+			// not using try_mutate here as we're not sure if key exits, and we're already doing a some value check on result of exists query check
+			match <AccountCurrencyTransferRestriction<T>>::get(account_id, currency_id) {
+				Some(allowance_count) if allowance_count <= 1 => {
+					<AccountCurrencyTransferRestriction<T>>::remove(account_id, currency_id);
+					Ok(())
+				}
+				Some(allowance_count) => {
+					// check in this case should not ever be needed
+					let new_allowance_count = allowance_count
+						.checked_sub(1)
+						.ok_or(Error::<T>::AllowanceCountArithmeticError)?;
```

Comment:
I think you can use `cfg_traits::ops::ensure_sub`. These methods will be part of Substrate from `0.9.37`, and allow you to eliminate `Overflow` errors from the pallet. 

Related info from "our" guidelines: https://centrifuge.hackmd.io/sLMxV5WoTy2q57fJTfbNLQ?view#Proposal

Reply:
👍 Updated to use ensure_sub

Reply:
There are some similar codes using `checked_add`,  maybe it makes sense to use `ensure_add` too there

Reply:
👍 updated the checked_add to use ensure_add as well :)